src/main/java/seedu/address/model/task/ScheduledTask.java

```diff
@@ -21,7 +21,8 @@
 
     protected ScheduledTask(Description description, TaskDateTime taskDateTime, ModuleCode moduleCode) {
         this.description = description;
-        this.moduleCode = moduleCode;
+        this.moduleCode = moduleCode.getClone(); // returns a defensive copy
```

Comment:
This is unnecessary defense as the `moduleCode` in the constructor is not going to be referred to in any other places.

Reply:
It may be useful just in case we do things with `Task` in `v2.0` features. It may prevent unnecessary change when we are implementing `Task mark as done` command too.

Reply:
not good example there -_-
Again, of all things, `ModuleCode` is the least necessary thing to go defensive
Actually, all classes are immutable so defensiveness is wasteful